fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalCatalogRelation.java

```diff
@@ -97,13 +98,19 @@ public List<String> getQualifier() {
      * Full qualified name parts, i.e., concat qualifier and name into a list.
      */
     public List<String> qualified() {
+        if (qualifier.size() == 3) {
+            return qualifier;
+        }
         return Utils.qualifiedNameParts(qualifier, table.getName());
     }
 
     /**
      * Full qualified table name, concat qualifier and name with `.` as separator.
      */
     public String qualifiedName() {
+        if (qualifier.size() == 3) {
```

Comment:
why we use special branch?

Reply:
qualifier may only contain db name or contain full name (catalog.db.table)，for the first case, need to concat the table name here, for the latter case, we don't need to.